loyalty_mass_mailing/models/loyalty_program.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from ast import literal_eval
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class LoyaltyProgram(models.Model):
+    _inherit = "loyalty.program"
+
+    mailing_ids = fields.One2many(
+        comodel_name="mailing.mailing",
+        inverse_name="program_id",
+        string="Mailings",
+        copy=False,
+    )
+    mailing_count = fields.Integer(
+        compute="_compute_mailing_count", string="Mailing count"
+    )
+
+    partner_applicability_domain = fields.Char(
+        compute="_compute_partner_applicability_domain",
+    )
+
+    @api.depends("rule_ids.rule_partners_domain")
+    def _compute_partner_applicability_domain(self):
+        for program in self:
+            partner_domains = [
+                literal_eval(domain)
+                for domain in program.rule_ids.mapped("rule_partners_domain")
+                if domain
+            ]
+            if all(partner_domains):
+                program.partner_applicability_domain = expression.OR(partner_domains)
```

Comment:
Shouldn't this be?

```suggestion
                program.partner_applicability_domain = expression.OR([partner_domains])
```

Reply:
Not necessary, here is an example of the value output in a test debug:
```
>>> partner_domains
[[('id', '=', 41)], [('id', '=', 42)]]

>>> program.partner_applicability_domain
"['|', ('id', '=', 41), ('id', '=', 42)]"
```